Importar numpy

In [ ]:
import numpy as np

Crear los arrays de precios y cantidades sin usar pandas    

In [ ]:
import numpy as np

cantidades = np.array([
    2, 15, 5, 10, 3, 20, 2, 8, 12, 1,
    4, 30, 5, 3, 25, 2, 6, 18, 12, 2,
    4, 30, 8, 3, 4, 20, 15, 6, 1, 10,
    3, 22, 5, 25, 2, 4, 12, 18, 2, 3,
    15, 5, 2, 3, 20, 6, 18, -1, 4, 25,
    3, 2, 20, 35, 5, 2, 7, 9, 20, 4
])

precios = np.array([
    1200, 25, 350, 85, 1200, 95, 350, 120, 95, 1200,
    350, 25, 85, 120, 95, 1200, 350, 95, 25, 1200,
    120, 95, 85, 350, 1200, 25, 95, 120, 1200, 85,
    350, 95, 1200, 25, 120, 350, 85, 95, 350, 1200,
    25, 120, 350, 1200, 95, 85, 25, 1200, 120, 95,
    350, 1200, 25, 95, 350, 1200, 120, 85, 95, 350
])

print(precios)
print(cantidades)

[1200   25  350   85 1200   95  350  120   95 1200  350   25   85  120
   95 1200  350   95   25 1200  120   95   85  350 1200   25   95  120
 1200   85  350   95 1200   25  120  350   85   95  350 1200   25  120
  350 1200   95   85   25 1200  120   95  350 1200   25   95  350 1200
  120   85   95  350]
[ 2 15  5 10  3 20  2  8 12  1  4 30  5  3 25  2  6 18 12  2  4 30  8  3
  4 20 15  6  1 10  3 22  5 25  2  4 12 18  2  3 15  5  2  3 20  6 18 -1
  4 25  3  2 20 35  5  2  7  9 20  4]


Verificar la longitud de los arrays para compararlo y confirmarlo con el csv

In [ ]:
print(len(cantidades))
print(len(precios))

60
60


Calcular el total de ingresos brutos usando el producto punto     

In [ ]:
ingresos = np.dot(cantidades,precios)
print(ingresos)

88205


Calcular la media, mediana, desviación estándar y percentiles 25 y 75 sobre el campo precio_unitario     

In [ ]:
media = np.mean(precios)
mediana= np.median(precios)
desv = np.std(precios)
perc = np.percentile(precios, [25,75])

print (f"La media es: {media}")
print (f"La mediana es: {mediana}")
print (f"La desviación estándar es: {desv}")
print (f"El percentil 25 es: {perc[0]} y el percentil 75 es: {perc[1]}")

La media es: 378.25
La mediana es: 120.0
La desviación estándar es: 445.0597572236789
El percentil 25 es: 92.5 y el percentil 75 es: 350.0


La media quiere decir que, en promedio, los productos tienen un precio de $378.35

La mediana muestra que el 50% de los productos tiene un precio menor o igual a $120, mientras que el otro 50% tiene un precio mayor o igual a este valor.

La desviación estándar es de 445.05, lo que evidencia una alta dispersión en los precios. Esto indica que existen diferencias significativas entre los productos de menor costo y los de mayor valor.

El 25% de los precios de los productos son menores o iguales a $92.5

El 75% de los precios de los productos son mayores o iguales a $350


Reflexión CRISP-DM

1. Entendimiento del negocio.
2. Entendimiento de datos.
3. Preparación de datos. 
4. Modelado.
5. Evaluación.
6. Despliegue.

Hasta esta parte del reto se tiene hasta la fase 2 Entendimiento de datos, siendo esta la más importante hasta el momento, debido a que se explora el dataset, y se crea manualmente unos arrays con los datos de precios y cantidades, es muy importante resaltar que esta exploración permite identificar que existe diferencias en los formatos en los que están presentados los datos, un ejemplo claro, es la fila 8, en la cual los datos están separados por coma, a diferencia de gran parte de los datos que se encuentran en columnas distintas, esta identificación permitió que esos datos no se tomaran como nulos al momento de crear el array, y que la longitud de datos sea coherente y la misma que esta en el dataset.